src/liballoc/str.rs

```diff
@@ -578,6 +579,6 @@ impl str {
 /// ```
 #[stable(feature = "str_box_extras", since = "1.20.0")]
 #[inline]
-pub unsafe fn from_boxed_utf8_unchecked(v: Box<[u8]>) -> Box<str> {
-    Box::from_raw(Box::into_raw(v) as *mut str)
+pub unsafe fn from_boxed_utf8_unchecked<A: Alloc + AllocHelper>(v: Box<[u8], A>) -> Box<str, A> {
```

Comment:
This is a breaking change, I'm pretty sure.

Reply:
How come?

Reply:
Actually, I'm not sure. I feel like there might be edgecases where it's not backwards compatible.

Reply:
[This](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=1267b5762faaaf2d932197085338cfd1) is an example of something similar where it's not backward compatible because type-inference isn't smart enough.

Reply:
I think that's acceptable. It's basically impossible to do anything with a higher order function whose arguments and return type are not constrained by the type checker.